### ISSUE TITLE
N5 receive mms

### ISSUE THREAD

**Issue by maximumhax** [ubports/ubuntu-touch#1673]
- Device: Nexus 5
- Channel: 16.04 ota15
- Build:

### Steps to reproduce

Open messages, have someone send mms photo

### Expected behavior

Receive the mms photo

### Actual behavior

Nothing happens

### Logfiles and additional information

I was really hoping this would have been corrected by now.. If i had this  i could actually use this as a daily driver.. Is there any phone that UT actually works on? Atleast the core stuff?

**Comment by lduboeuf** [comment]
see here for the long story : https://forums.ubports.com/topic/5100/the-mms-lost-story/89

btw you should be able at least to have an alert message, isn't it the case ?

**Comment by maximumhax** [comment]
Nope, I don't get anything. No way of knowing if someone even tried sending
me a photo.. I was talking to my next door neighbor last night testing and
I sent her a photo, she got it, she sent one, I had to ask if she sent it
as I didn't know. I was so stoked when I read there was a new version.. but
the 1 thing keeping me from using this daily still not fixed.. I guess I'll
wait until the next version.

**Comment by lduboeuf** [comment]
humm strange if you're able to send you should be able to receive, ( mms needs to be enable in messaging-app settings of course ). You can see if it is workng by sending mms to yourself

**Comment by ZikZakFR** [comment]
Same issue.
It may work for a few days then not working any longer.
When I attach a picture from file or camera it takes forever to get it displayed, then it gets stuck in the sending animation and won't ever be sent.
When in this state I can't receive any and no alert, so I can't even tell I was supposed to get a MMS.

I noticed that by changing update channel and so installing the new (or old) version with the reboot it works again (for a few hours or days).

When it happens, in dmesg I see many denied_mask=r for profile="messaging-app" coming from apparmor
Could it be the reason?

**Comment by lduboeuf** [comment]
@ZikZakFR I think this is another issue, you are talking about sending a MMS.
Something might be wrong in your APN settings ?
I remember my N5 had issues when cellular signal wzs weak

**Comment by lduboeuf** [comment]
@ZikZakFR what do you mean by old/new version. Do you mean stable/ development channel ?

**Comment by lduboeuf** [comment]
can you sent a MMS to yourself ?

**Comment by ZikZakFR** [comment]
The issue in my case is detected because my message isn't leaving, when I asked the other party I then discovered I never received their messages.

Exactly, jumping from Stable to Candidate, then back to Stable. No APN modification.

Sending a MMS to self faces the same issue, stuck into an endless sending message.

**Comment by lduboeuf** [comment]
weird issue. idk what can cause the delta between the switch between channels. I do that quite frequently without issue

**Comment by ZikZakFR** [comment]
Just upgraded from Candidate to Edge and I sent to self the same picture as above, received less than 10 seconds later.
I'll keep it on 'edge' until the issue occurs again and when it does I'll check if dmesg reports again apparmor denying "messaging-app"; for now dmesg is quite clean (apart from some rate limit exceeded).

**Comment by lduboeuf** [comment]
I didn't know there srill an edge channel ready to use. There is a Release Candidate coming, hope it will work again for you

**Comment by ZikZakFR** [comment]
And the next day my MMS doesn't go, Apparmor is again denying messaging-app according to dmesg.

Then Edge update is available, auick update and reboot, previous MMS still stuck so I cancel it and send a brand new one successfully.

**Comment by lduboeuf** [comment]
Could you test it on last RC ?
But someone with a N5 must confirm, because there so much possible cause. ( hardware, APN setting, bad signal ) 
Can you provide Apparmor issues you see ?

**Comment by ZikZakFR** [comment]
Difficult to get the pictures out of the phone, it is no longer recognized as a storage for my Linux distro. I'll find a way.
I just installed RC 137.

[Screenshot of dmesg when MMS can't be sent](https://www.verybad.link/dmesg.png)

**Comment by lduboeuf** [comment]
The best would be to provide nuntium logs `/home/phablet/.cache/upstart/nuntium.log` when the issue occured

**Comment by lduboeuf** [comment]
I would recommend also a full reinstall with the installer

**Comment by ZikZakFR** [comment]
I'd prefer not to, as I would loose my MFA configured on this smartphone.
For nuntium I'll take a note of that, never heard of that before.

**Comment by lduboeuf** [comment]
@ZikZakFR i mean reinstall with the installer without wiping
`nuntium` is the service responsible for sending/retrieving MMS

**Comment by ZikZakFR** [comment]
And MMS is dead again today.
[Here](https://www.verybad.link/nuntium.log.1.gz) are the latest logs from nuntium.

**Comment by lduboeuf** [comment]
@ZikZakFR , the logs shows that you received and sent MMSes, i don't see an issue here.

So , this could be more a rendering issue.
To make sure, if the download, or upload of the file went fine,  you can see the history database ( located in  `~/.local/share/history-service/history.sqlite`
there should be entries in `text_event_attachments` table and the field `filePath` points to the attachments path where they should be physically here.

**Comment by ZikZakFR** [comment]
And today, gone again. 
I tried re-setting the APN to no avail.

**Comment by lduboeuf** [comment]
What are the steps to have MMS gone ?. Did you installed anything, switched to another branch ? Anything else ?
Does your APN settings persist after reboot ?

**Comment by ZikZakFR** [comment]
Nothing special, power on the Nexus5 sends a picture via MMS and then I can tell immediately something is wrong s the message will take dozen of seconds to actually shows that it is in a sending loop that will never complete.
APN never changed. Today I switched back to Stable in order to make the MMS work again.
The next day it fails I'll switch to another channel, rinse and repeat every time it happens as I haven't found another workaround for many months.

**Comment by lduboeuf** [comment]
I remember that the N5 was not very stable for cellular signals. To confirm are you on 4G ?

**Comment by lduboeuf** [comment]
and please report logs when something don't work.
Notice that you can test by sending a MMS to yourself

**Comment by ZikZakFR** [comment]
I use only 3G network, I never got MMS to work properly on 4G.
Moreover I only activate Data when I require it (otherwise the battery depletes too quickly).
Sending a MMS to others or to self ends the same way, what is more concerning is that until I try to send I will never know that it stopped working as reception of a MMS will never be notified, it will simply never happened. I can't tell how many MMS were never delivered due to this bug.

The N5 on the contrary has been stable for many years, I do not know when this bug started to happen but then I discovered here that there was a bug already open, so I shared my experience about the same without opening another ticket.

**Comment by lduboeuf** [comment]
humm, as we see in nuntium logs, you have many mms that were never acknowledged, showing that there is issues with probably signals. I suspect hardware first ( cable antenna can be the issue or modem ). 
But if it is software, then hard reset would be best and make sure firmware is up to date too.
I asked on channel Fr, no one reported issues on N5. 
An easy thing to test is moving to 4G first
